src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEM.cpp

```diff
@@ -234,71 +251,97 @@ void AcousticWaveEquationSEM::precomputeSourceAndReceiverTerm( MeshLevel & mesh
       } );
     } );
   } );
+
+  EventBase const & eventbase = this->getGroupByPath< EventBase >( "/Problem/Events/solverApplications" );
+  real64 const & dt = eventbase.getReference< real64 >( EventBase::viewKeyStruct::forceDtString() );
+
+  forAll< EXEC_POLICY >( m_sourceValue.size( 1 ), [=] GEOSX_HOST_DEVICE ( localIndex const isrc )
+  {
+    for( localIndex cycle = 0; cycle < m_sourceValue.size( 0 ); ++cycle )
+    {
+      real64 time = cycle*dt;
+      m_sourceValue[cycle][isrc] = evaluateRicker( time, this->m_timeSourceFrequency, this->m_rickerOrder );
+    }
+  } );
```

Comment:
```suggestion
  real64 const timeSourceFrequency = this->m_timeSourceFrequency;
  real64 const rickerOrder = this->m_rickerOrder;
  arrayView2d< real64 > const sourceValue = m_sourceValue.toView();
  forAll< EXEC_POLICY >( m_sourceValue.size( 1 ), [=] GEOSX_HOST_DEVICE ( localIndex const isrc )
  {
    for( localIndex cycle = 0; cycle < sourceValue.size( 0 ); ++cycle )
    {
      real64 time = cycle*dt;
      sourceValue[cycle][isrc] = evaluateRicker( time, timeSourceFrequency, rickerOrder );
    }
  } );
```
Above, I suggest some modifications to make sure that the code compiles and works on the GPU. In general, you cannot pass `this`, `array2d`, or class data members to the GPU kernels. You have to pass views on the data instead.